pandas/tseries/frequencies.py

```diff
@@ -386,51 +386,76 @@ def get_period_alias(offset_str):
 _INVALID_FREQ_ERROR = "Invalid frequency: {0}"
 
 
-def to_offset(freqstr):
+def to_offset(freq):
```

Comment:
this is technically a change, e.g. someone passes by name.

so we can either change it back or deprecate it (deprecating it is fine). you can use the ``deprecate`` decorator